### PR TITLE
Resource 234 aws api gateway request validator

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill the box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_api_gateway_request_validator.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  example "
+    describe aws_api_gateway_request_validator(rest_api_id: 'REST_API_ID', request_validator_id: 'REQUEST_VALIDATOR_ID') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters(required: %i(rest_api_id request_validator_id))
+    raise ArgumentError, "#{@__resource_name__}: rest_api_id must be provided" unless opts[:rest_api_id] && !opts[:rest_api_id].empty?
```
Comment (sathish-progress) [inline libraries/aws_api_gateway_request_validator.rb:18]:
```suggestion
    raise ArgumentError, "#{@__resource_name__}: rest_api_id must be provided" if opts[:rest_api_id].blank?
```

File: libraries/aws_api_gateway_request_validator.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @display_name = opts[:request_validator_id]
+    catch_aws_errors do
+      resp = @aws.apigateway_client.get_request_validator({ rest_api_id: opts[:rest_api_id], request_validator_id: opts[:request_validator_id] })
+      @res = resp.to_h
+      create_resource_methods(@res)
+    end
+  end
+
+  def exists?
+    !@res.nil? && !@res.empty?
```
Comment (sathish-progress) [inline libraries/aws_api_gateway_request_validator.rb:29]: @res.blank?

File: libraries/aws_api_gateway_request_validator.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  def request_validator_id
+    return nil unless exists?
+    @res[:request_validator_id]
+  end
```
Comment (sathish-progress) [inline libraries/aws_api_gateway_request_validator.rb:35]: not needed!

File: libraries/aws_api_gateway_request_validators.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  example "
+    describe aws_api_gateway_request_validators(rest_api_id: 'REST_API_ID') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters(required: %i(rest_api_id))
+    raise ArgumentError, "#{@__resource_name__}: rest_api_id must be provided" unless opts[:rest_api_id] && !opts[:rest_api_id].empty?
```
Comment (sathish-progress) [inline libraries/aws_api_gateway_request_validators.rb:18]: use `.blank?`

**Review by sathish-progress — APPROVED** [review]

**Review by dkumaras — APPROVED** [review]

**Pull request merged**